provider-example/src/sign/ecdsa.rs

```diff
@@ -0,0 +1,58 @@
+use pkcs8::DecodePrivateKey;
+use pki_types::PrivateKeyDer;
+use rustls::{
+    sign::{Signer, SigningKey},
+    SignatureAlgorithm, SignatureScheme,
+};
+use signature::RandomizedSigner;
+use signature::SignatureEncoding;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub struct EcdsaSigningKeyP256 {
+    key: Arc<p256::ecdsa::SigningKey>,
+    scheme: SignatureScheme,
+}
+
+impl TryFrom<PrivateKeyDer<'_>> for EcdsaSigningKeyP256 {
+    type Error = pkcs8::Error;
+
+    fn try_from(value: PrivateKeyDer<'_>) -> Result<Self, Self::Error> {
+        match value {
+            PrivateKeyDer::Pkcs8(der) => {
+                p256::ecdsa::SigningKey::from_pkcs8_der(der.secret_pkcs8_der()).map(|kp| Self {
+                    key: Arc::new(kp),
+                    scheme: SignatureScheme::ECDSA_NISTP256_SHA256,
+                })
+            }
+            _ => todo!(),
```

Comment:
Maybe something like:
```suggestion
            _ => panic!("unsupported private key format"),
```